coalib/parsing/DefaultArgParser.py

```diff
@@ -22,6 +23,23 @@ def _format_action_invocation(self, action):
             return ', '.join(parts)
 
 
+class PathArg(str):
+    """
+    Uni(xi)fying OS-native directory separators in path arguments.
+
+    Removing the pain from interactively using coala in a Windows cmdline,
+    because backslashes are interpreted as escaping syntax and therefore
+    removed when arguments are turned into coala settings
+
+    >>> import os
+    >>> PathArg(os.path.join('path', 'with', 'separators'))
+    'path/with/separators'
+    """
+
+    def __new__(cls, path):
+        return str.__new__(cls, path.replace(os.path.sep, '/'))
```

Comment:
why not using `__init__`?

Reply:
or generally a conversion function instead of a type

Reply:
@Makman2 _\*whisper\* ;) Let's silently forget your `__init__` question ;) `str` is immutable. So `__init__` can't change it's value anymore. So the only way to override it's construction is, well, by overriding the actual constructor method, which is actually `__new__`_

Function or class? Well, you won't believe me how long I though about that... But you have to make a decision :D In the end I just thought a class looks more like a `type=` than a function ;)

Reply:
> *whisper* ;) Let's silently forget your __init__ question ;) str is immutable. So __init__ can't change it's value anymore. So the only way to override it's construction is, well, by overriding the actual constructor method, which is actually __new__

Already thought so it has to do with str's immutability, but just wanted to clarify, I actually forgot nearly everything about python's metaclassing magic 😅 

I would just say for the sake of simplicity use a function, however we might want to change str representations to be nice and cool, and a function can't do that ;) so let's keep it then :)